sdk/tests/SerializationAndDeserializationTests.cc

```diff
@@ -0,0 +1,199 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#include "AccountId.h"
+#include "ContractId.h"
+#include "ExchangeRate.h"
+#include "ExchangeRateSet.h"
+#include "Hbar.h"
+#include "TransactionId.h"
+#include "TransactionReceipt.h"
+#include "Transfer.h"
+
+#include "impl/TimestampConverter.h"
+
+#include <chrono>
+#include <gtest/gtest.h>
+#include <proto/basic_types.pb.h>
+#include <proto/exchange_rate.pb.h>
+
+using namespace Hedera;
+
+class SerializationAndDeserializationTests : public ::testing::Test
+{
+protected:
+  [[nodiscard]] inline const uint64_t getTestShardNum() const { return mShardNum; }
+  [[nodiscard]] inline const uint64_t getTestRealmNum() const { return mRealmNum; }
+  [[nodiscard]] inline const uint64_t getTestContractNum() const { return mContractNum; }
+  [[nodiscard]] inline const AccountId& getTestAccountId() const { return mAccountId; }
+
+private:
+  const uint64_t mShardNum = 1;
+  const uint64_t mRealmNum = 2;
+  const uint64_t mContractNum = 3;
+  const AccountId mAccountId = AccountId(10ULL);
+  const std::chrono::system_clock::time_point mValidStart = std::chrono::system_clock::time_point();
+};
+
+TEST_F(SerializationAndDeserializationTests, SerializeAccountIdToProtobufTest)
+{
+  // Given
+  const std::string testAccountIdStr = "111.222.333";
+  AccountId testAccountId(testAccountIdStr);
+
+  // When
+  auto protoAccountId = std::unique_ptr<proto::AccountID>(testAccountId.toProtobuf());
+
+  // Then
+  EXPECT_EQ(protoAccountId->shardnum(), testAccountId.getShardNum());
+  EXPECT_EQ(protoAccountId->realmnum(), testAccountId.getRealmNum());
+  EXPECT_EQ(protoAccountId->accountnum(), testAccountId.getAccountNum());
+}
+
+TEST_F(SerializationAndDeserializationTests, DeserializeAccountIdFromProtobufTest)
+{
+  // Given
+  proto::AccountID testProtoAccountId = proto::AccountID();
+  testProtoAccountId.set_shardnum(123);
+  testProtoAccountId.set_realmnum(456);
+  testProtoAccountId.set_accountnum(789);
+
+  // When
+  AccountId accountId = AccountId::fromProtobuf(testProtoAccountId);
+  
+  // Then
+  EXPECT_EQ(accountId.toString(), "123.456.789");
+}
+
+TEST_F(SerializationAndDeserializationTests, SerializeContractIdТoProtobufTest)
+{
+  // Given
+  uint64_t testShardNum = getTestShardNum();
+  uint64_t testRealmNum = getTestRealmNum();
+  uint64_t testContractNum = getTestContractNum();
+  ContractId testContractId = ContractId(testShardNum, testRealmNum, testContractNum);
+
+  // When
+  auto protoContractId = std::unique_ptr<proto::ContractID>(testContractId.toProtobuf());
+
+  // Then
+  EXPECT_EQ(protoContractId->shardnum(), testShardNum);
+  EXPECT_EQ(protoContractId->realmnum(), testRealmNum);
+  EXPECT_EQ(protoContractId->contractnum(), testContractNum);
+}
+
+TEST_F(SerializationAndDeserializationTests, DeserializeContractIdFromProtobufTest)
```

Comment:
To write this test I used test [ProtobufContractId](https://github.com/hashgraph/hedera-sdk-cpp/blob/c54139e8bc8f507e1f08d052b15eec3d188a0bd0/sdk/tests/ContractIdTest.cc#L76) from ContractIdTest.

Reply:
Test `DeserializeContractIdFromProtobufTest` moved into `ContractIdTest.cc` in commit https://github.com/hashgraph/hedera-sdk-cpp/pull/176/commits/42c4b74b470cbb18745489aebed04b87c7b75235.